### PR TITLE
For Issue-33: [CMake] Refresh all the CMakeLists.txt and target dependency.

### PR DESCRIPTION
**Pull request by HuaYuan-Tseng**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,94 +1,133 @@
...
+target_sources(Model
+    PRIVATE
+    ${MODEL_SOURCES}
+    PUBLIC
+    ${MODEL_HEADERS}
+)
+target_include_directories(Model
+    PUBLIC
+    ${CMAKE_SOURCE_DIR}/models
+    ${CMAKE_SOURCE_DIR}/models/card
```
Comment (YingChen-Lee) [inline CMakeLists.txt:58]: I'm thinking maybe we can just use `target_include_directories(Model PUBLIC ${CMAKE_SOURCE_DIR}/models)`.
In this way, in Model layer, we can use `models` as the root directory for the includes like `#include "card/card.h"` or `#include "hand.h"`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,94 +1,133 @@
...
+    ${CMAKE_SOURCE_DIR}/models/events
+)
+
+# Controller2Usecase Library
+
+add_library(Controller2Usecase STATIC "")
```
Comment (YingChen-Lee) [inline CMakeLists.txt:64]: Since this library link the Model library, maybe `Controller2Domain` would be a better name?

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,94 +1,133 @@
...
+    PRIVATE
+    ${MODEL_SOURCES}
+    PUBLIC
+    ${MODEL_HEADERS}
+)
+target_include_directories(Model
+    PUBLIC
+    ${CMAKE_SOURCE_DIR}/models
+    ${CMAKE_SOURCE_DIR}/models/card
+    ${CMAKE_SOURCE_DIR}/models/events
```
Comment (YingChen-Lee) [inline CMakeLists.txt:59]: I'm thinking maybe we can just use `target_include_directories(Model PUBLIC ${CMAKE_SOURCE_DIR}/models)`.
In this way, in Model layer, we can use `models` as the root directory for the includes like `#include "card/card.h"` or `#include "hand.h"`.

**Comment by YingChen-Lee** [comment]
Excellent!!!

File: CMakeLists.txt (hunk 3 above)
Comment (HuaYuan-Tseng) [reply CMakeLists.txt:59]: Nice idea !!!

Comment (YingChen-Lee) [reply CMakeLists.txt:59]: Or you can leave it as it is. I can update this when updating the `#includes`.

File: CMakeLists.txt (hunk 2 above)
Comment (HuaYuan-Tseng) [reply CMakeLists.txt:64]: I need other people's opinions. 🥹

File: CMakeLists.txt (hunk 3 above)
Comment (HuaYuan-Tseng) [reply CMakeLists.txt:59]: I will think twice again.

**Pull request merged**